server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -363,16 +375,33 @@ public NodeGatewayStartedShards(StreamInput in) throws IOException {
             } else {
                 storeException = null;
             }
+            if (in.getVersion().onOrAfter(Version.V_3_0_0) && in.readBoolean()) {
```

Comment:
Keeping version check to `3.0.0` as to ensure bwcTest exercise this code for `2.3.0` code.